api/views.py

```diff
@@ -14,11 +15,21 @@ def analyze_address(request):
             radius = data.get("radius")
 
             if not street:
```

Comment:
Suggestion for the future - there are libraries like [pydantic](https://docs.pydantic.dev/latest/) that will validate input from arbitrary sources into nice, Python classes. It's usually a good idea to use them instead of doing validation on your own.